tests/php/WidgetsTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Tests;
+
+use Bolt\Snippet\HtmlInjector;
+use Bolt\Snippet\QueueProcessor;
+use Bolt\Snippet\RequestZone;
+use Bolt\Snippet\Target;
+use Bolt\Widget\BoltHeaderWidget;
+use Bolt\Widget\SnippetWidget;
+use Bolt\Widget\WeatherWidget;
+use Bolt\Widgets;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Twig\Environment;
+use Twig\Loader\ArrayLoader;
+
+class WidgetsTest extends TestCase
+{
+    public function testProcessWidgetsInQueue(): void
+    {
+        $queueprocessor = new QueueProcessor(new HtmlInjector());
+        $requestStack = new RequestStack();
+        $requestStack->push(Request::createFromGlobals());
+
+        $loader = new ArrayLoader(['weather.twig' => '[Hello, weather!]']);
+        $twig = new Environment($loader);
+
+        $widgets = new Widgets($requestStack, $queueprocessor, $twig);
+        $response = new Response('<html><body>foo</body></html>');
+
+        $snippet = (new SnippetWidget())
```

Comment:
if you want to create widgets with setters chaining, how about using a builder instead?
```
        $snippet = (new WidgetBuilder(SnippetWidget::class))
            ->setTemplate('*foo*')
            ->setTarget(Target::END_OF_BODY)
			->getWidget();
```
Then we can finally remove setters from them.
(it can be created in separate PR)

Reply:
Ok, let's do that in separate PR. Added #408